pt/lessons/ecto/associations.md

```diff
@@ -0,0 +1,505 @@
+para tornar as associações de personagens de um filme disponíveis. Isso é o que nos permite realizar chamadas como `movie.characters`.
+
+#### A migração Belongs
+
+Agora nós estamos prontos para construir nossa migração e `schema` para `Character`. Um personagem pertence(`belongs to`) a um filme, então vamos definir uma migração que especifique o relacionamento.
+
+Primeiro, precisamos gerar a migração:
+
+```console
+mix ecto.gen.migration create_characters
+```
+
+Para declarar que um personagem pertence a um filme, precisamos da tabela `characters` e que ela possua uma coluna `movie_id`. Nós queremos que essa coluna funcione como uma chave estrangeira. Podemos alcançar isso com a seguinte linha, na chamada para `create_table/1`:
+
+```elixir
+add :movie_id, references(:movies)
+```
+
+Assim, nossa migração deve ser algo como:
+
+```elixir
+# priv/migrations/*_create_characters.exs
+defmodule Example.Repo.Migrations.CreateCharacters do
+  use Ecto.Migration
+
+  def change do
+    create_table(:characters) do
+      add :name, :string
+      add :movie_id, references(:movies)
+    end
+  end
+end
+```
+
+#### O Schema Belongs To
+
+Nosso esquema precisa definir a relação `belongs to` entre um personagem e seu filme,
+
+```elixir
+# lib/example/character.ex
+
+defmodule Example.Character do
+  use Ecto.Schema
+
+  schema "characters" do
+    field :name, :string
+    belongs_to :movie, Example.Movie
+  end
+end
+```
+
+Vamos dar uma olhada mais a fundo na macro `belongs_to/3`. Ao invés de adicionar a coluna `movie_id` na tabela `characters`, essa macro _não_ adiciona nada ao banco de dados. Ela nos _permite_ acessar os esquemas de `movies` associados através de `characters`. Ela utiliza a chave estrangeira `movie_id` para tornar os `characters` associados com um dado filme quando executamos a consulta sobre os personagens. Isso nos permite chamar `character.movie`.
+
+Agora nós estamos prontos para executar as migrações:
+
+```console
+mix ecto.migrate
+```
+
+### Belong To/Has One
+
+Digamos que um filme tenha um distribuidor. Por exemplo, o Netflix é o distribuidor do filme original "Bright".
+
+Vamos definir a migração e o esquema `Distributor` com o relacionamento " belongs to ". Primeiro, é preciso gerar a migração:
+
+```console
+mix ecto.gen.migration create_distributors
+```
+
+Nossa migração deve adicionar uma chave estrangeira de `movie_id` à tabela` distributors`:
+
+```elixir
+# priv/repo/migrations/*_create_distributors.exs
+
+defmodule Example.Repo.Migrations.CreateDistributors do
+  use Ecto.Migration
+
+  def change do
+    create table(:distributors) do
+      add :name, :string
+      add :movie_id, references(:movies)
+    end
+  end
+end
+```
+
+E o esquema `Distributor` deve usar a macro` belongs_to/3` para nos permitir chamar `distributor.movie` e procurar o filme associado de um distribuidor usando esta chave estrangeira.
```

Comment:
WDYT about using `à` instead of `de um` here? Seems to better denoted the meaning of a movie associated to a distributor.

```diff
- e procurar o filme associado de um distribuidor usando esta chave estrangeira.
+ e procurar o filme associado à distribuidor usando esta chave estrangeira.
```

Reply:
I think the correct would be `a um distribuidor` without "`":

```suggestion
E o esquema `Distributor` deve usar a macro` belongs_to/3` para nos permitir chamar `distributor.movie` e procurar o filme associado a um distribuidor usando esta chave estrangeira.
```